### PR TITLE
Add uploading models method for MODAC client

### PR DESCRIPTION
**Pull request by andriialbatov**

### CODE REVIEW

**Comment by andriialbatov** [comment]
Screenshot 2024-08-01 at 4 27 51 PM

I thought at first it started to upload, it took some time after I ran the upload_file method before it failed

File: atomsci/modac/__init__.py [hunk 1]
```diff
@@ -136,3 +165,46 @@ def download_all_files_in_collection(
...
+
+        _logger.warning(f"Uploading file to {url}")
+        try:
+            with requests.put(url, files=files, headers=self._token_headers()) as resp:
+                resp.raise_for_status()
+                _logger.info(f"File uploaded successfully: {resp.json()}")
+                return True
+        except requests.exceptions.RequestException as e:
+            _logger.error(f"File upload failed: {e}")
+            raise
```
Comment (Kartstig) [inline atomsci/modac/__init__.py]:
```suggestion
            raise e
```
It's good practice to raise the previous exception unless you have a custom one to raise

We could make our own custom exception for this module but that might not be necessary yet

File: atomsci/modac/__init__.py [hunk 2]
```diff
@@ -136,3 +165,46 @@ def download_all_files_in_collection(
...
+
+        files = {
+            "dataObjectRegistration": (
+                "attributesFile.json",
+                json.dumps(attributes_file),
+                "application/json",
+            ),
+            "dataObject": (
+                os.path.basename(data_file_path),
+                open(data_file_path, "rb"),
```
Comment (Kartstig) [inline atomsci/modac/__init__.py]: I think your code works for 99.99% of the time, but let's say `_logger.warning` encountered a runtime error. This file would remain open. You were correct in using `finally` to catch for most error cases. However, I think it's just best practice to use the `with open(...) as f` syntax, and then you don't have to worry or think about closing:


```python
with open(data_file_path, "rb") as f:
    ...
```

**Review by Kartstig — CHANGES_REQUESTED** [review]
I'd like to fix the code around file closing

**Review by Kartstig — APPROVED** [review]

**Pull request merged**